names/names.py

```diff
@@ -10,11 +11,35 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
-duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+# NESTED FOR LOOPS
+
+# duplicates = []                           O(1)
+# for name_1 in names_1:                    O(n)
+#     for name_2 in names_2:                O(n)
+#         if name_1 == name_2:              O(1)
+#             duplicates.append(name_1)     O(1)
+
+# TOTAL                                     O(2**n)
+# nested for runtime    6.037697076797485 seconds
+
+
+# BINARY SEARCH TREE
+
+duplicates = []                             # O(1)
+# Set up one list in binary search tree
+bst = BinarySearchTree(names_2[0])          # O(1)
+for name_2 in names_2[1:]:                  # O(n)
+    bst.insert(name_2)                      # O(1)
+
+# Loop through next list and use bst to see
+# if it is a duplicate. If so append to
```

Comment:
also try not to use too much white space in your code, if you can set your tabs to 2 indents/spaces